### PR TITLE
Adding retry logic to delete

### PR DESCRIPTION
**Pull request by swiencki**
### Which issue this PR addresses:

Relates to PR https://github.com/Azure/ARO-RP/pull/1447

### What this PR does / why we need it:

Retry logic on delete for E2E failures

### Test plan for issue:

Testing using e2e pipelines

### Is there any documentation that needs to be updated for this PR?

Comments in code

### CODE REVIEW

File: pkg/cluster/delete.go [hunk 1]
```diff
@@ -284,8 +284,15 @@ func (m *manager) Delete(ctx context.Context) error {
...
-	err = m.deleteResources(ctx)
+	// Retry logic due to race condition between PLS and nic
+	// TODO: Follow up on false future return
+	for retry := 0; retry < 3; retry++ {
+		err = m.deleteResources(ctx)
+		if err == nil {
+			break
+		}
+		m.log.Errorf("Failed to delete %s", err.Error())
+	}
```
Comment (troy0820) [inline pkg/cluster/delete.go:295]: I wouldn't delete the log entry 
I'm wondering if using `k8s.io/apimachinery/pkg/util/wait` function `PollImmediateUntil` would be valuable.  We used the same logic `pkg/util/cluster/cluster.go` when fixing up network security groups.

Comment (jewzaam) [reply pkg/cluster/delete.go:296]: Talking internal, agree this is good, example cited: https://github.com/Azure/ARO-RP/blob/81229ae1968ebc9f09be4d14428d19d8dde7cce3/pkg/util/cluster/cluster.go#L506-L513

File: pkg/cluster/delete.go [hunk 2]
```diff
@@ -284,8 +284,16 @@ func (m *manager) Delete(ctx context.Context) error {
 		return err
 	}
 
+	// Retry logic due to race condition between PLS and nic
+	// TODO: Follow up on false future return
```
Comment (jim-minter) [inline pkg/cluster/delete.go:288]: explain more clearly what's happening, add incident reference here

File: pkg/cluster/delete.go [hunk 3]
```diff
@@ -284,8 +284,16 @@ func (m *manager) Delete(ctx context.Context) error {
...
 	m.log.Printf("deleting resources")
-	err = m.deleteResources(ctx)
+	for retry := 0; retry < 3; retry++ {
+		err = m.deleteResources(ctx)
+		if err == nil {
+			break
+		}
+		m.log.Errorf("Failed to delete %s", err.Error())
```
Comment (jim-minter) [inline pkg/cluster/delete.go:295]:
```suggestion
		m.log.Info(err)
		time.Sleep(30 * time.Second)
```

**Review by jim-minter — COMMENTED** [review]
Yuck, but I agree it's the right thing to do.  Couple of suggestions:

File: pkg/cluster/delete.go (hunk 1 above)
Comment (swiencki) [reply pkg/cluster/delete.go]: Consensus is to go with the temp retry logic. Please see comments in code for more details.

File: pkg/cluster/delete.go (hunk 2 above)
Comment (swiencki) [reply pkg/cluster/delete.go]: // PR 1483 is a temporary fix
	// PLS failing to delete its own managed NIC causing E2E failures
	// TODO: Remove PR 1483 changes on resolution of 239789689

File: pkg/cluster/delete.go (hunk 3 above)
Comment (swiencki) [reply pkg/cluster/delete.go]: Resolved. Thank you

**Comment by bryanro92** [comment]
This adds some technical debt to our code base, but the retry logic should fix regression in PLS service. Once our ticket to NRP team is resolved we can verify their fix through the removal of this retry.

**Review by mjudeikis — APPROVED** [review]

**Review by gvanderpotte — COMMENTED** [review]
LGTM

**Pull request merged**